packages/program-boilerplate/src/index.js

```diff
@@ -27,68 +28,112 @@ export {
+*
+* @example webtask ({
+*          "AFTER_USER_CREATED_OR_UPDATED" : handleUserUpsert,
+*          "AFTER_USER_EVENT_PROCESSED": handleUserEvent,
+*          "REFERRAL": handleReferralTrigger,
+*          "PROGRAM_INTROSPECTION": handleIntrospection
+*           })
+* @param {Object} handlers - Key-value pairs, where key is a ProgramTriggerType (see {@link ProgramTriggerTypes}) or "PROGRAM_INTROSPECTION";
+* ProgramTrigger handlers must accept a transaction instance as parameter.
+* Program-Introspection must accept a template as parameter and returns a new template.
+* @returns {function} - A function that fits in the webtask programming model. See {@link https://webtask.io/docs/model}.
+*
+*
+*/
 export function webtask(handlers = {}) {
-    /**
-     * A function that fits in the webtask programming model. 
-     * If the messageType of context is 
-     * (a) "PROGRAM_INTROSPECTION":
-     * The handler must take the current template and rules as parameter, and returns the modified template.
-     * (b) "PROGRAM_TRIGGER":
-     * It creates a Transaction instance and passes it to a program handler.
-     * The handler will operate on the transaction instance to generate mutations analytics, which, together with programId are passed as 
-     * the result parameter of the callback function.
-     * 
-     * @param {WebtaskContext} context - A webtask context with several properties.
-     * @param {requestCallback} cb - A callback function.To indicate completion, the function must call the callback with two arguments: an error, and the result.
-     */
-    return function (context, cb) {
-        switch (context.body.messageType || "PROGRAM_TRIGGER") {
-            case "PROGRAM_INTROSPECTION":
-                const template = context.body.template;
-                const rules = context.body.rules;
-                const program = context.body.program;
-                // Make modifications to template based on rules here if necessary.
-                // ...
-                const handleIntrospection = handlers["PROGRAM_INTROSPECTION"];
-                const newTemplate = handleIntrospection && (handleIntrospection(template,rules,program) || handleIntrospection(template,rules)) || template;
-                cb(null, newTemplate);
-                break;
-            case "PROGRAM_TRIGGER": 
-                let transaction = new Transaction(context);
-                const triggerType = context.body.activeTrigger.type;
-                const handleTrigger = handlers[triggerType];
-                if (handleTrigger) {
-                    handleTrigger(transaction);
-                }  
-                cb(null, transaction.toJson());
-                break;
-            default:
-                cb(null, {});
-                break;
+  const express = require('express')();
+  const bodyParser = require('body-parser');
+  const compression = require('compression');
+
+  express.use(bodyParser.json());
+  express.use(compression());
+
```

Comment:
We should add validation into the express server to ensure it only responds to secure (https) requests. This will ensure program developers don't accidentally point to an insecure endpoint, which could expose customer data.